start-tests.sh

```diff
@@ -7,106 +7,267 @@
+  fi
 
-if [[ -z $OPTIONS_MANAGED_BASEDOMAIN || -z $OPTIONS_MANAGED_USER || -z $OPTIONS_MANAGED_PASSWORD ]]; then
-   log_color "yellow" "One or more variables are undefined. Copying kubeconfigs...\n"
-   cp /opt/.kube/import-kubeconfig ./config/import-kubeconfig
-   echo $OPTIONS_KUBECONFIG_MOUNT_PATH
+  export CYPRESS_OPTIONS_HUB_KUBECONFIG=$OPTIONS_HUB_KUBECONFIG
+fi
+
+VERSION=`oc get subscriptions.operators.coreos.com -A -o yaml | grep currentCSV:\ advanced-cluster-management | awk '{$1=$1};1' | sed "s/currentCSV:\ advanced-cluster-management.v//"`
+log_color "green" "Testing with ACM Version": "$VERSION\n"
+
+log_color "cyan" "Checking RedisGraph deployment."
+installNamespace=`oc get subscriptions.operators.coreos.com --all-namespaces | grep advanced-cluster-management | awk '{print $1}'`
+rgstatus=`oc get srcho searchoperator -o jsonpath="{.status.deployredisgraph}" -n ${installNamespace}`
+
+if [ "$rgstatus" == "true" ]; then
+  echo -e "RedisGraph deployment is enabled.\n"
+else
+  echo -e "RedisGraph deployment disabled, enabling and waiting 60 seconds for the search-redisgraph-0 pod.\n"
+  oc set env deploy search-operator DEPLOY_REDISGRAPH="true" -n $installNamespace
+  sleep 60
+fi
+
+# Search for managed clusters.
+MANAGED_CLUSTERS=($(oc get managedclusters -o custom-columns='name:.metadata.name' --no-headers))
+
+# Check to see if there are any managed cluster available.
+if [[ ${#MANAGED_CLUSTERS[@]} == "1" && ${MANAGED_CLUSTERS[0]} =~ "local-cluster" ]]; then
+  echo -e "No managable clusters detected for the hub cluster: $CYPRESS_OPTIONS_HUB_BASEDOMAIN.\n"
+  export CYPRESS_SKIP_MANAGED_CLUSTER_TEST=true
 else
-  log_color "cyan" "Logging into the managed cluster using credentials and generating the kubeconfig..."
-  mkdir ./import-kubeconfig && touch ./import-kubeconfig/kubeconfig
-  export KUBECONFIG=$(pwd)/import-kubeconfig/kubeconfig
-  export OPTIONS_MANAGED_URL="https://api.$OPTIONS_MANAGED_BASEDOMAIN:6443"
-  oc login --server=$OPTIONS_MANAGED_URL -u $OPTIONS_MANAGED_USER -p $OPTIONS_MANAGED_PASSWORD --insecure-skip-tls-verify
-  unset KUBECONFIG
-  log_color "yellow" "Copying managed cluster kubeconfig to ./cypress/config/import-kubeconfig ...\n"
-  cp ./import-kubeconfig/* ./config/import-kubeconfig
+  echo -e "Detected clusters within the fleet: ${GREEN}${MANAGED_CLUSTERS[@]}${NC}\n"
+
+  if [[ -z $OPTIONS_MANAGED_BASEDOMAIN || -z $OPTIONS_MANAGED_USER || -z $OPTIONS_MANAGED_PASSWORD ]]; then
+    log_color "red" "One or more variables are undefined for imported cluster." "(set ${PURPLE}OPTIONS_MANAGED_BASEDOMAIN, OPTIONS_MANAGED_BASEDOMAIN, and OPTIONS_MANAGED_BASEDOMAIN${NC} to execute the test with environment variables)\n"
+
+    # The mount path will be set by CICD.
+    if [[ ! -z $OPTIONS_KUBECONFIG_MOUNT_PATH ]]; then
+      echo -e "Detected kubeconfig mount path for imported cluster at: $OPTIONS_KUBECONFIG_MOUNT_PATH"
+      OPTIONS_MANAGED_KUBECONFIG=$OPTIONS_KUBECONFIG_MOUNT_PATH
+    fi
```

Comment:
This is a variable set by CICD. If the path isn't exported or set, we'll check the `default` or `OPTIONS_MANAGED_KUBECONFIG` path for the `kubeconfig` file.